cuda/private/macros/cuda_test.bzl

```diff
@@ -0,0 +1,25 @@
+load("//cuda/private:rules/cuda_library.bzl", "cuda_library")
+
+def cuda_test(
+        name,
+        srcs = [],
+        copts = [],
+        visibility = ["//visibility:private"],
+        data = [],
+        deps = []):
+    cuda_library_name = "_" + name
+
+    cuda_library(
+        name = cuda_library_name,
+        srcs = srcs,
+        copts = copts,
+        deps = deps,
+    )
+
+    native.cc_test(
+        name = name,
+        copts = copts,
+        visibility = visibility,
+        data = data,
+        deps = [cuda_library_name],
+    )
```

Comment:
What about all the other attributes for cc_test? In particular exec_comaptible_with and exec_properties are important for selecting the correct execution platform as the user needs to ensure this is run on a platform that has a cuda compatible GPU present.
The easiest way to do this is to forward kwargs to this rule so you don't have to explicitly list all attributes.